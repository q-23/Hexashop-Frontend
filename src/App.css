.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.fa-stack[data-count]:after{
  position:absolute;
  right:0%;
  top:1%;
  content: attr(data-count);
  font-size:30%;
  padding:.6em;
  border-radius:999px;
  line-height:.75em;
  color: white;
  background:rgba(255,0,0,.85);
  text-align:center;
  min-width:2em;
  font-weight:bold;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
