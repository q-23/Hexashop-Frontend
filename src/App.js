import React from 'react';
import {Switch, Route} from 'react-router-dom';

import Container from "./components/_Shared/Container";
import Header from "./components/Header";
import Menu from "./components/Menu";
import { routes } from "./routes";
import {MOCK_CATEGORIES as mockCategories} from "./components/Menu/Menu.mock.data";


function App() {

  return (
    <Container main mx_auto>
      <Header/>
      <section className={'app'}>
        <Menu categories={mockCategories}/>
        <Container>
          <Switch>
            {
              routes.map((route, idx) =>
                <Route exact={route.exact} path={route.path} component={route.component} key={route.path + idx}/>
              )
            }
          </Switch>
        </Container>
      </section>
    </Container>
  );
}

export default App;
