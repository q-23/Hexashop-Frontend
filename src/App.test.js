import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { MemoryRouter as Router } from 'react-router-dom';

test('renders learn react link', () => {
  const wrapper = render(
    <Router keyLength={0}>
      <App />
    </Router>
    );
  expect(wrapper).toMatchSnapshot();
});
