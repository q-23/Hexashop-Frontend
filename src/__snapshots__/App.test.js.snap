// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm hvlQDu"
      >
        <header>
          <div
            class="sc-AxirZ iuAETO"
          >
            <div
              class="sc-AxiKw iWDEz"
            >
              <i
                class="sc-AxhUy edbuyE fa fa-github"
              />
            </div>
            <div
              class="sc-AxiKw cPJjoP"
            >
              <img
                alt="logo"
                class="sc-AxhCb kGbSP"
                src="logo.png"
              />
            </div>
            <div
              class="sc-AxiKw iWDEz"
            >
              <i
                class="sc-AxhUy dQzhvk fa fa-shopping-bag"
              />
              <i
                class="sc-AxhUy edbuyE fa fa-search"
              />
            </div>
          </div>
          <hr />
        </header>
        <section
          class="app"
        >
          <nav>
            <ul>
              <ul
                class="sc-AxheI jlxAAw"
              >
                <a
                  class="sc-AxgMl dZDZAF"
                  href="/"
                >
                  Start
                </a>
              </ul>
              <ul
                class="sc-AxheI jlxAAw"
              >
                <a
                  class="sc-AxgMl dZDZAF"
                  href="/shoes"
                >
                  Shoes
                </a>
              </ul>
              <ul
                class="sc-AxheI jlxAAw"
              >
                <a
                  class="sc-AxgMl dZDZAF"
                  href="/jackets"
                >
                  Jackets
                </a>
              </ul>
              <ul
                class="sc-AxheI jlxAAw"
              >
                <a
                  class="sc-AxgMl dZDZAF"
                  href="/pants"
                >
                  Pants
                </a>
              </ul>
              <ul
                class="sc-AxheI jlxAAw"
              >
                <a
                  class="sc-AxgMl dZDZAF"
                  href="/caps"
                >
                  Caps
                </a>
              </ul>
            </ul>
          </nav>
          <div
            class="sc-AxjAm cXrawr"
          >
            <div />
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm hvlQDu"
    >
      <header>
        <div
          class="sc-AxirZ iuAETO"
        >
          <div
            class="sc-AxiKw iWDEz"
          >
            <i
              class="sc-AxhUy edbuyE fa fa-github"
            />
          </div>
          <div
            class="sc-AxiKw cPJjoP"
          >
            <img
              alt="logo"
              class="sc-AxhCb kGbSP"
              src="logo.png"
            />
          </div>
          <div
            class="sc-AxiKw iWDEz"
          >
            <i
              class="sc-AxhUy dQzhvk fa fa-shopping-bag"
            />
            <i
              class="sc-AxhUy edbuyE fa fa-search"
            />
          </div>
        </div>
        <hr />
      </header>
      <section
        class="app"
      >
        <nav>
          <ul>
            <ul
              class="sc-AxheI jlxAAw"
            >
              <a
                class="sc-AxgMl dZDZAF"
                href="/"
              >
                Start
              </a>
            </ul>
            <ul
              class="sc-AxheI jlxAAw"
            >
              <a
                class="sc-AxgMl dZDZAF"
                href="/shoes"
              >
                Shoes
              </a>
            </ul>
            <ul
              class="sc-AxheI jlxAAw"
            >
              <a
                class="sc-AxgMl dZDZAF"
                href="/jackets"
              >
                Jackets
              </a>
            </ul>
            <ul
              class="sc-AxheI jlxAAw"
            >
              <a
                class="sc-AxgMl dZDZAF"
                href="/pants"
              >
                Pants
              </a>
            </ul>
            <ul
              class="sc-AxheI jlxAAw"
            >
              <a
                class="sc-AxgMl dZDZAF"
                href="/caps"
              >
                Caps
              </a>
            </ul>
          </ul>
        </nav>
        <div
          class="sc-AxjAm cXrawr"
        >
          <div />
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
